tests/phpunit/tests/rest-api/rest-block-type-controller.php

```diff
@@ -0,0 +1,404 @@
+		$data     = $response->get_data();
+		$this->assertCount( 1, $data );
+		$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block_name );
+		$this->check_block_type_object( $block_type, $data[0], $data[0]['_links'] );
+	}
+
+	/**
+	 * @ticket 47620
+	 */
+	public function test_get_item() {
+		$block_name = 'fake/test';
+		wp_set_current_user( self::$admin_id );
+		$request    = new WP_REST_Request( 'GET', '/wp/v2/block-types/' . $block_name );
+		$response   = rest_get_server()->dispatch( $request );
+		$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block_name );
+		$this->check_block_type_object( $block_type, $response->get_data(), $response->get_links() );
+	}
+
+	/**
+	 * @ticket 47620
+	 */
+	public function test_get_item_with_styles() {
+		$block_name   = 'fake/styles';
+		$block_styles = array(
+			'name'         => 'fancy-quote',
+			'label'        => 'Fancy Quote',
+			'style_handle' => 'myguten-style',
+		);
+		register_block_type( $block_name );
+		register_block_style( $block_name, $block_styles );
+		wp_set_current_user( self::$admin_id );
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/block-types/' . $block_name );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertEqualSets( array( $block_styles ), $data['styles'] );
+
+	}
+
+	/**
+	 * @ticket 47620
+	 */
+	public function test_get_item_with_styles_merge() {
+		$block_name   = 'fake/styles2';
+		$block_styles = array(
+			'name'         => 'fancy-quote',
+			'label'        => 'Fancy Quote',
+			'style_handle' => 'myguten-style',
+		);
+		$settings     = array(
+			'styles' => array(
+				array(
+					'name'         => 'blue-quote',
+					'label'        => 'Blue Quote',
+					'style_handle' => 'myguten-style',
+				),
+			),
+		);
+		register_block_type( $block_name, $settings );
+		register_block_style( $block_name, $block_styles );
+		wp_set_current_user( self::$admin_id );
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/block-types/' . $block_name );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$expected = array(
+			array(
+				'name'         => 'fancy-quote',
+				'label'        => 'Fancy Quote',
+				'style_handle' => 'myguten-style',
+			),
+			array(
+				'name'         => 'blue-quote',
+				'label'        => 'Blue Quote',
+				'style_handle' => 'myguten-style',
+			),
+		);
+		$this->assertEqualSets( $expected, $data['styles'] );
+
+	}
+
+	/**
+	 * @ticket 47620
+	 */
+	public function test_get_block_invalid_name() {
+		$block_type = 'fake/block';
+		wp_set_current_user( self::$admin_id );
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/block-types/' . $block_type );
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertErrorResponse( 'rest_block_type_invalid', $response, 404 );
+	}
+
+	/**
+	 * @ticket 47620
+	 */
+	public function test_get_item_invalid() {
```

Comment:
@TimothyBJacobs Should I extend this test with all fields then?

Reply:
I think that'd be valuable, as well as a test with the explicit empty/default values.

Reply:
More tests added.